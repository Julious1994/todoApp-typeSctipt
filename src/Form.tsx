import * as React from "react";
import { HandleOnChangeInterface } from './commonInterface';

interface formProps {
  text?: string;
  onClick(): void;
  onChange(e: HandleOnChangeInterface): void;
}

const Form = (props: formProps) => (
  <div>
    <input value={props.text} type="text" onChange={props.onChange} />
    <button onClick={props.onClick}>Add todo</button>
  </div>
);

Form.defaultProps = {
  text: ""
};

export default Form;