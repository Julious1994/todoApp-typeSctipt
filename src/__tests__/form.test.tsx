import * as React from 'react';
import { shallow } from 'enzyme';
import Form from '../Form';

let wrapper: any;
const onChangeMock = jest.fn();
const onClickMock = jest.fn();

beforeEach(() => {
  const props = {
    onChange: onChangeMock,
    onClick: onClickMock,
    todo: {
      text: '',
    },
  };
  wrapper = shallow(<Form {...props} />);
});

it('should render form view', () => {
  const input = wrapper.find('input');
  const button = wrapper.find('button');
  expect(input.length).toBe(1);
  expect(button.length).toBe(1);
  expect(wrapper).toMatchSnapshot();
});

test('button title', () => {
  const button = wrapper.find('button').at(0);
  expect(button.text()).toBe('Add todo');
});
