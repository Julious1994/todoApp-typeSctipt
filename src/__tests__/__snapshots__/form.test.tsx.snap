// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render form view 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    onChange={[MockFunction]}
    onClick={[MockFunction]}
    text=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[MockFunction]}
          type="text"
          value=""
        />,
        <button
          onClick={[MockFunction]}
        >
          Add todo
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [MockFunction],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add todo",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Add todo",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[MockFunction]}
            type="text"
            value=""
          />,
          <button
            onClick={[MockFunction]}
          >
            Add todo
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [MockFunction],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add todo",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Add todo",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
